pkg/framework/plugins/removefailedpods/failedpods.go

```diff
@@ -0,0 +1,154 @@
+	}
+
+	var includedNamespaces, excludedNamespaces sets.String
+	if failedPodsArgs.Namespaces != nil {
+		includedNamespaces = sets.NewString(failedPodsArgs.Namespaces.Include...)
+		excludedNamespaces = sets.NewString(failedPodsArgs.Namespaces.Exclude...)
+	}
+
+	podFilter, err := podutil.NewOptions().
+		WithFilter(handle.Evictor().Filter).
+		WithNamespaces(includedNamespaces).
+		WithoutNamespaces(excludedNamespaces).
+		WithLabelSelector(failedPodsArgs.LabelSelector).
+		BuildFilterFunc()
+	if err != nil {
+		return nil, fmt.Errorf("error initializing pod filter function: %v", err)
+	}
+
+	podFilter = podutil.WrapFilterFuncs(func(pod *v1.Pod) bool { return pod.Status.Phase == v1.PodFailed }, podFilter)
+
+	if len(failedPodsArgs.ExcludeOwnerKinds) > 0 {
+		excludeOwnerKinds := sets.NewString(failedPodsArgs.ExcludeOwnerKinds...)
+		podFilter = podutil.WrapFilterFuncs(podFilter, func(pod *v1.Pod) bool {
+			ownerRefList := podutil.OwnerRef(pod)
+			for _, owner := range ownerRefList {
+				if excludeOwnerKinds.Has(owner.Kind) {
+					return false
+				}
+			}
+			return true
+		})
+	}
+
+	if failedPodsArgs.MinPodLifetimeSeconds != nil {
+		minPodLifetimeSeconds := *failedPodsArgs.MinPodLifetimeSeconds
+		podFilter = podutil.WrapFilterFuncs(podFilter, func(pod *v1.Pod) bool {
+			podAgeSeconds := uint(metav1.Now().Sub(pod.GetCreationTimestamp().Local()).Seconds())
+			return podAgeSeconds > minPodLifetimeSeconds
+		})
+	}
+
+	if len(failedPodsArgs.Reasons) > 0 {
+		failedPodsReasons := sets.NewString(failedPodsArgs.Reasons...)
+		podFilter = podutil.WrapFilterFuncs(podFilter, func(pod *v1.Pod) bool {
+			reasons := getFailedContainerStatusReasons(pod.Status.ContainerStatuses)
+
+			if pod.Status.Phase == v1.PodFailed && pod.Status.Reason != "" {
+				reasons = append(reasons, pod.Status.Reason)
+			}
+
+			if failedPodsArgs.IncludingInitContainers {
+				reasons = append(reasons, getFailedContainerStatusReasons(pod.Status.InitContainerStatuses)...)
+			}
+
+			return failedPodsReasons.HasAny(reasons...)
+		})
+	}
+
+	return &RemoveFailedPods{
+		handle:    handle,
+		podFilter: podFilter,
+		args:      failedPodsArgs,
+	}, nil
+}
+
+// Name retrieves the plugin name
+func (d *RemoveFailedPods) Name() string {
+	return PluginName
+}
+
+// Deschedule extension point implementation for the plugin
+func (d *RemoveFailedPods) Deschedule(ctx context.Context, nodes []*v1.Node) *framework.Status {
+	for _, node := range nodes {
+		klog.V(1).InfoS("Processing node", "node", klog.KObj(node))
+		pods, err := podutil.ListAllPodsOnANode(node.Name, d.handle.GetPodsAssignedToNodeFunc(), d.podFilter)
+		if err != nil {
+			// no pods evicted as error encountered retrieving evictable Pods
+			return &framework.Status{
+				Err: fmt.Errorf("error listing pods on a node: %v", err),
+			}
+		}
+		totalPods := len(pods)
+		for i := 0; i < totalPods; i++ {
+			d.handle.Evictor().Evict(ctx, pods[i], evictions.EvictOptions{})
+		}
```

Comment:
Do we want to

```
if podEvictor.NodeLimitExceeded(node) {
	continue
}
```

here?

Reply:
Good catch, all the rebases and merge conflicts are making this harder to maintain